test/fuzz/test_runner.py

```diff
@@ -178,21 +178,32 @@ def generate_corpus_seeds(*, max_jobs, build_dir, seed_dir, targets):
     """
     logging.info("Generating corpus seeds to {}".format(seed_dir))
 
-    jobs = []
-    for target in targets:
+    def generate_job():
+        target = random.choice(targets)
+        mutate_depth = random.randint(1, 15)
+        use_value_profile = 0
+        if random.randint(1, 10) == 1:
+            use_value_profile = 1
+        max_len = 2**random.randint(6, 12)
+        if random.randint(1, 20) == 1:
+            max_len = 0
         target_seed_dir = os.path.join(seed_dir, target)
         os.makedirs(target_seed_dir, exist_ok=True)
         args = [
             os.path.join(build_dir, 'src', 'test', 'fuzz', 'fuzz'),
-            "-runs=100000",
+            "-max_total_time=1800",
+            "-reload=0",
+            "-mutate_depth={}".format(mutate_depth),
+            "-use_value_profile={}".format(use_value_profile),
+            "-max_len={}".format(max_len),
             target_seed_dir,
         ]
-        jobs.append([target, args])
+        return [target, args]
 
     running_jobs = set()
-    while len(jobs) > 0 or len(running_jobs) > 0:
-        if len(running_jobs) < max_jobs and len(jobs) > 0:
-            target, args = jobs.pop()
+    while True:
```

Comment:
It was nice to be able to put the `while true` outside the script. The benefit was that the outside could update this script, the code repo, as well as the seeds repo. This allows continuous fuzzing of the master branch with the latest code and seeds while atomically updating the build system and this script.

Reply:
@MarcoFalke I don't think there is a problem with that really, as no progress is lost by interrupting. Just stop the test_runner at any point, update/recompile, and run again.

Reply:
That would complicate the outside script because it had to poll for changes in the repo, while running the fuzzer script. As soon as there are changes, interrupt and rerun.

Reply:
I y'all like the `while True`, maybe expose it via some kind of option?

Reply:
It's not so much the while True I like, but the fact that this doesn't leave cores unused after most of the jobs are done. Doing it randomly also more easily accommodates adding more configurations (which presumably would hurt your workflow equally much, as every new combination would multiply the time between checks by some factor).

What about adding an option to make this automatically exit after some specified wall clock time?